### PR TITLE
Rate-limit + timeout fix for incomplete socket reads

### PR DESCRIPTION
**Pull request by massand**
There are two parts to the fix:
1. ensure request content read is timed out within 5 seconds to avoid content-length being set to greater value than actual content 
2. rate-limit all clients to 10 concurrent requests per uid per service (this applies uniformly to all uids)

### CODE REVIEW

File: http-common/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,8 @@ openssl = { version = "0.10", optional = true }
 percent-encoding = "2"
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
-tokio = { version = "1", features = ["net", "rt-multi-thread"], optional = true }
+tokio = { version = "1", features = ["net", "rt-multi-thread", "sync"], optional = true }
+tokio-io-timeout = { version = "1" }
```
Comment (arsing) [inline http-common/Cargo.toml:24]: `tokio-io-timeout` by itself is not enough. That sets the limit for every individual read, not for the whole read of the request. That means the client can drip-feed one byte every 5s and the request won't time out.

You also need to apply a [`tokio::time::Timeout`](https://docs.rs/tokio/1.5.0/tokio/time/fn.timeout.html) on all the calls to `hyper::body::to_bytes(body)` inside the `make_service` macro.

**Review by arsing — CHANGES_REQUESTED** [review]

File: http-common/src/connector.rs [hunk 2]
```diff
@@ -27,8 +27,15 @@ pub enum AsyncStream {
...
 #[derive(Debug)]
 pub enum Incoming {
-    Tcp(tokio::net::TcpListener),
-    Unix(tokio::net::UnixListener),
+    Tcp {
+        listener: tokio::net::TcpListener,
+    },
+    Unix {
+        listener: tokio::net::UnixListener,
+        user_state: futures_util::lock::Mutex<
```
Comment (arsing) [inline http-common/src/connector.rs:35]: This doesn't need to be an async mutex. Just use `std::sync::Mutex`.

File: http-common/Cargo.toml (hunk 1 above)
Comment (massand) [reply http-common/Cargo.toml:24]: Fixed

File: http-common/src/connector.rs (hunk 2 above)
Comment (massand) [reply http-common/src/connector.rs]: Fixed

File: http-common/src/connector.rs [hunk 3]
```diff
@@ -80,8 +80,8 @@ impl Incoming {
                 let (unix_stream, _) = listener.accept().await?;
                 let ucred = unix_stream.peer_cred()?;
                 let user_state = user_state
-                    .lock()
-                    .await
+                    .get_mut()
```
Comment (arsing) [inline http-common/src/connector.rs:83]: Oh, this already receives `&mut self`, so even a `std::sync::Mutex` is unnecessary. You can just have `user_state: BTreeMap` directly.

Comment (massand) [reply http-common/src/connector.rs]: Good catch. Fixed.

**Review by arsing — APPROVED** [review]

**Pull request merged**